sdk/storage/src/queue/requests/set_queue_metadata_builder.rs

```diff
@@ -86,7 +87,7 @@ where
             url.as_str(),
             &http::Method::PUT,
             &|mut request| {
-                request = AddAsHeader::add_as_header(self.client_request_id(), request);
+                request = add_header(self.client_request_id(), request);
                 request = AddAsHeader::add_as_header(&self.metadata(), request);
```

Comment:
Why not this one as well?

Reply:
`metadata` is not `Option` so we cannot use the helper function. Should we create a non option variant? I would be a one liner so I thought better to call the trait function directly.

Reply:
It's up to you. In cosmos I wrap the ones that are not options in `Some`. The compiler most likely can see through this and will skip the if check `add_header`.

Reply:
I think than using wrapping into a fake `Option` for the sake of saving a function makes for confusing code. 

I suggest we have two functions instead: `add_optional_header` and `add_mandatory_header`. Their names are self-explanatory even to a casual reader.